dev/devicelab/bin/tasks/smoke_test_build_test.dart

```diff
@@ -14,18 +14,27 @@ Future<void> main(List<String> args) async {
 }
 
 class FakeBuildTestTask extends BuildTestTask {
-  FakeBuildTestTask(List<String> args) : super(args, runFlutterClean: false) {
-    deviceOperatingSystem = DeviceOperatingSystem.fake;
-  }
+  FakeBuildTestTask(List<String> args) : super(args, runFlutterClean: false);
 
   @override
   // In prod, tasks always run some unit of work and the test framework assumes
   // there will be some work done when managing the isolate. To fake this, add a delay.
-  Future<void> build() => Future<void>.delayed(const Duration(milliseconds: 500));
+  Future<void> build() async {
+    await Future<void>.delayed(const Duration(milliseconds: 500));
+
+    if (targetPlatform != DeviceOperatingSystem.fake) {
+      throw Exception('Only DeviceOperatingSystem.fake is supported');
+    }
```

Comment:
Can you check `targetPlatform` before the 500ms delay to fail faster?

Reply:
(I see you had that originally, wondering what's up with the "fix async timing"?)

Reply:
I'm looking into this. When running all the tests where the test runner fails to connect to the isolate, but when I run each test file individually it runs fine. I think the issue is in `framework.dart` or `runner.dart`